src/main/java/org/springframework/data/domain/Pageable.java

```diff
@@ -161,4 +161,20 @@ default Optional<Pageable> toOptional() {
 		return isUnpaged() ? Optional.empty() : Optional.of(this);
 	}
 
+	/**
+	 * Returns an {@link OffsetCursorRequest} from this pageable if the page request {@link #isPaged() is paged}.
+	 *
+	 * @return
+	 * @throws IllegalStateException if the request is {@link #isUnpaged()}
+	 * @since 3.1
+	 */
+	default CursorRequest toCursorRequest() {
+
+		if (isUnpaged()) {
+			return CursorRequest.uncursored();
+		}
+
+		return OffsetCursorRequest.ofSize(getPageSize(), getSort()).withOffset(getOffset());
+	}
+
```

Comment:
The created `CursorRequest` points to page 0 and element offset 0 regardless of the page number. Shouldn't `Pageable.getOffset()` be passed into `ofSize`?